modules/ecs/cluster/main.tf

```diff
@@ -4,23 +4,45 @@ resource "aws_ecs_cluster" "ecs_cluster" {
   lifecycle {
     create_before_destroy = true
   }
-}
 
-resource "aws_iam_role_policy" "ecs_instance_role_policy" {
-  name   = "ecs-instance-role-policy-${var.environment}"
-  policy = file("${path.module}/templates/ecs-instance-role-policy.json")
-  role   = aws_iam_role.ecs_role.id
+  tags = var.tags
 }
 
-resource "aws_iam_role" "ecs_role" {
-  name               = "ecs-instance-role-${var.environment}"
-  assume_role_policy = file("${path.module}/templates/ecs-role.json")
+#resource "aws_iam_role_policy" "ecs_instance_role_policy" {
+#  name   = "ecs-instance-role-policy-${var.environment}"
+#  policy = file("${path.module}/templates/ecs-instance-role-policy.json")
+#  role   = aws_iam_role.ecs_role.id
+#}
+#
+#resource "aws_iam_role" "ecs_role" {
+#  name               = "ecs-instance-role-${var.environment}"
+#  assume_role_policy = file("${path.module}/templates/ecs-role.json")
+#}
+#
+#resource "aws_iam_instance_profile" "ecs_instance_profile" {
+#  name = "ecs-instance-profile-${var.environment}"
+#  path = "/"
+#  role = aws_iam_role.ecs_role.name
+#}
+
+
+
+resource "aws_iam_role" "ecs_runtime" {
+  name               = "${var.app_name}-${var.role_name}"
+  assume_role_policy = data.aws_iam_policy_document.assume_ecs_role.json
+
+  tags = var.tags
 }
 
-resource "aws_iam_instance_profile" "ecs_instance_profile" {
-  name = "ecs-instance-profile-${var.environment}"
-  path = "/"
-  role = aws_iam_role.ecs_role.name
+data "aws_iam_policy_document" "assume_ecs_role" {
+  statement {
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      identifiers = ["ecs-tasks.amazonaws.com", "ecs.amazonaws.com"]
```

Comment:
`ecs-role.json` has one additional service which I dont think is needed `ssm.amazonaws.com` why not just update that?